### PR TITLE
Update supported PHP versions

### PR DESCRIPTION
**Pull request by alcaeus**
|      Q       |   A
|------------- | -----------
| Type         | improvement
| BC Break     | no
| Fixed issues | 

#### Summary

* ~Drop PHP 7.1~ skipped for now to help out Symfony users
* Require build for PHP 7.4 to pass

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -28,13 +28,15 @@ jobs:
...
       env: DEPENDENCIES=low
+      php: 7.2
       install:
         - rm composer.lock
         - travis_retry composer update --prefer-dist --prefer-lowest
         - ./download-box.sh
 
     - stage: Test
       env: COVERAGE
+      php: 7.3
```
Comment (carusogabriel) [inline .travis.yml]: Why not `7.2` like the other jobs?

File: .travis.yml [hunk 2]
```diff
@@ -23,19 +22,18 @@ script:
   - ./vendor/bin/phpunit
 
 jobs:
-  allow_failures:
-    - php: 7.4snapshot
-
   include:
     - stage: Test
       env: DEPENDENCIES=low
+      php: 7.2
```
Comment (lcobucci) [inline .travis.yml]: Not needed since `7.2` is the lowest version already.

File: .travis.yml [hunk 3]
```diff
@@ -23,19 +22,18 @@ script:
...
       install:
         - rm composer.lock
         - travis_retry composer update --prefer-dist --prefer-lowest
         - ./download-box.sh
 
     - stage: Test
       env: COVERAGE
+      php: 7.3
```
Comment (lcobucci) [inline .travis.yml]: What's the need for this?

File: .travis.yml [hunk 4]
```diff
@@ -47,10 +45,12 @@ jobs:
 
     - stage: Code Quality
       env: CODING_STANDARDS
+      php: 7.2
```
Comment (lcobucci) [inline .travis.yml]: Not needed since `7.2` is the lowest version already.

File: .travis.yml [hunk 5]
```diff
@@ -47,10 +45,12 @@ jobs:
...
       install: travis_retry composer install --prefer-dist
       script: ./vendor/bin/phpcs
 
     - stage: Code Quality
       env: STATIC_ANALYSIS
+      php: 7.2
```
Comment (lcobucci) [inline .travis.yml]: Not needed since `7.2` is the lowest version already.

File: .travis.yml (hunk 1 above)
Comment (alcaeus) [reply .travis.yml]: Because typo :(

File: .travis.yml (hunk 2 above)
Comment (alcaeus) [reply .travis.yml]: Wasn't sure what version it picks, but if we can be sure it's 7.2 I'll drop it.

File: composer.lock [hunk 6]
```diff
@@ -4059,7 +4073,7 @@
     "prefer-stable": false,
     "prefer-lowest": false,
     "platform": {
-        "php": "^7.1"
+        "php": "^7.2"
```
Comment (lcobucci) [inline composer.lock]: This should be `^7.1` then :+1:

**Pull request merged**